common/lib/xmodule/xmodule/tabs.py

```diff
@@ -191,22 +205,88 @@ def from_json(tab_dict):
             InvalidTabsException if the given tab doesn't have the right keys.
         """
         # TODO: don't import openedx capabilities from common
-        from openedx.core.djangoapps.course_views.course_views import CourseViewTypeManager
+        from openedx.core.lib.course_tabs import CourseTabPluginManager
         tab_type_name = tab_dict.get('type')
         if tab_type_name is None:
             log.error('No type included in tab_dict: %r', tab_dict)
             return None
         try:
-            tab_type = CourseViewTypeManager.get_plugin(tab_type_name)
+            tab_type = CourseTabPluginManager.get_plugin(tab_type_name)
         except PluginError:
             log.exception(
                 "Unknown tab type %r Known types: %r.",
                 tab_type_name,
-                CourseViewTypeManager.get_course_view_types()
+                CourseTabPluginManager.get_tab_types()
             )
             return None
+
         tab_type.validate(tab_dict)
-        return tab_type.create_tab(tab_dict=tab_dict)
+        return tab_type(tab_dict=tab_dict)
+
+
+class StaticTab(CourseTab):
+    """
+    A custom tab.
+    """
+    type = 'static_tab'
+    is_default = False    # A static tab is never added to a course by default
+    allow_multiple = True
+
+    def __init__(self, tab_dict=None, name=None, url_slug=None):
+        def link_func(course, reverse_func):
+            """ Returns a url for a given course and reverse function. """
+            return reverse_func(self.type, args=[course.id.to_deprecated_string(), self.url_slug])
+
+        self.url_slug = tab_dict['url_slug'] if tab_dict else url_slug
+
+        if tab_dict is None:
+            tab_dict = dict()
+
+        if name is not None:
+            tab_dict['name'] = name
+
+        tab_dict['link_func'] = link_func
+        tab_dict['tab_id'] = 'static_tab_{0}'.format(self.url_slug)
+
+        super(StaticTab, self).__init__(tab_dict)
+
+    @classmethod
+    def is_enabled(cls, course, user=None):  # pylint: disable=unused-argument
+        """
+        Static tabs are viewable to everyone, even anonymous users.
+        """
+        return True
+
+    @classmethod
+    def validate(cls, tab_dict, raise_error=True):
+        """
+        Ensures that the specified tab_dict is valid.
+        """
+        return (super(StaticTab, cls).validate(tab_dict, raise_error)
+                and key_checker(['name', 'url_slug'])(tab_dict, raise_error))
+
+    def __getitem__(self, key):
+        if key == 'url_slug':
+            return self.url_slug
+        else:
+            return super(StaticTab, self).__getitem__(key)
```

Comment:
I'm asking why we're overriding this method. By default, calling `static_tab.url_slug` on an instance of `StaticTab` would return self.url_slug. calling `static_tab.some_property` would find `some_property` on static_tab which could have been either set by the child class or the parent class.